src/state/opamSysInteract.ml

```diff
@@ -1093,14 +1105,50 @@ let install_packages_commands_t ?(env=OpamVariable.Map.empty) config sys_package
     [`AsUser (Commands.msys2 config),
      "-Su"::"--noconfirm"::packages], None
   | Netbsd -> [`AsAdmin "pkgin", yes ["-y"] ("install" :: packages)], None
+  | Nix ->
+    (match switch with
+    | None ->
+        log "Nix depext must be passed switch";
+        [], None
+    | Some switch ->
+    let dir = OpamPath.Switch.meta OpamStateConfig.(!r.root_dir) switch in
+    let drvFile = OpamFilename.create dir (OpamFilename.basename (OpamFilename.raw "env.nix")) in
```

Comment:
At least `nix.env` needs to be added in OpamPath as it is used in another place in the code.

Reply:
Right!

Reply:
Done in https://github.com/ocaml/opam/pull/5982/commits/241d29cbc3594c9bee6126f29d713c32eb425c22 (pushed as a fixup commit)